nats/rtt_command.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func configureRTTCommand(app *kingpin.Application) {
+	app.Command("rtt", "Compute round-trip time to NATS server").Action(rtt)
+}
+
+func rtt(pc *kingpin.ParseContext) error {
+	if !strings.Contains(servers, "://") {
+		servers = fmt.Sprintf("nats://%s", servers)
+	}
+
+	u, err := url.Parse(servers)
+	if err != nil {
+		return err
+	}
+
+	if net.ParseIP(u.Hostname()) == nil {
+		addrs, _ := net.LookupHost(u.Hostname())
+		if len(addrs) <= 1 {
+			goto doSingle
```

Comment:
why have special handling for single ones? If we calculate the list of names->ips upfront we can just loop it once, if its one its one and if its more than one its more than one, same code paths.

This will remove some duplication in the printfs and ensure that the output format is consistent, I think we should also support outputing the the data as JSON

Reply:
If I enter a hostname that resolves to only 1 or enter a single IP I liked this better. I was running it quite a bit between setups here, demo and NGS.